mlrun/package/packagers_manager.py

```diff
@@ -0,0 +1,632 @@
+            logger.warn(
+                f"{artifact_key} was originally packaged by a packager of type '{packager_name}' but it was not "
+                f"found. Custom packagers should be added to the project running the function. MLRun will try using "
+                f"the packager '{packager.__name__}' instead. It is recommended to use the same packager for packing "
+                f"and unpacking to reduce unexpected behaviours."
+            )
+
+        # Unpack:
+        return packager.unpack(
+            data_item=data_item,
+            artifact_type=artifact_type,
+            instructions=instructions,
+        )
+
+    def link_packages(
+        self,
+        additional_artifacts: typing.List[Artifact],
+        additional_results: dict,
+    ):
+        """
+        Link packages between each other according to the provided extra data and metrics spec keys. A future link is
+        marked with ellipses (...). If no link was found, None will be used and a warning will get printed.
+
+        :param additional_artifacts: Additional artifacts to link (should come from a `mlrun.MLClientCtx`).
+        :param additional_results:   Additional results to link (should come from a `mlrun.MLClientCtx`).
+        """
+        # Join the manager's artifacts and results with the additional ones to look for a link in all of them:
+        joined_artifacts = [*additional_artifacts, *self.artifacts]
+        joined_results = {**additional_results, **self.results}
+
+        # Go over the artifacts and link:
+        for artifact in self.artifacts:
+            # Go over the extra data keys:
+            for key in artifact.spec.extra_data:
+                # Future link is marked with ellipses:
+                if artifact.spec.extra_data[key] is ...:
+                    # Look for an artifact or result with this key to link it:
+                    extra_data = self._look_for_extra_data(
+                        key=key, artifacts=joined_artifacts, results=joined_results
+                    )
+                    # Print a warning if a link is missing:
+                    if extra_data is None:
+                        logger.warn(
+                            f"Could not find {key} to link as extra data for {artifact.key}."
+                        )
+                    # Link it (None will be used in case it was not found):
+                    artifact.spec.extra_data[key] = extra_data
+            # Go over the metrics keys (if available):
+            if hasattr(artifact.spec, "metrics"):
+                for key in artifact.spec.metrics:
+                    # Future link is marked with ellipses:
+                    if artifact.spec.metrics[key] is ...:
+                        # Link it (None will be used in case it was not found):
+                        artifact.spec.metrics[key] = joined_results.get(key, None)
+
+    class _InstructionsNotesKeys:
```

Comment:
```suggestion
    class _InstructionsNotesKey(Enum):
```

Reply:
I see it as a collection of available keys to use, this is why its plural. Regarding the Enum, its just an inner class of dictionary keys for inner use, so I'm not sure its necessary.